File: src/net/airvantage/model/AvSystem.java
package net.airvantage.model;

import java.util.Collection;
import java.util.Map;


public class AvSystem {
    public String uid;
    public String name;
    public String commStatus;
    public Long lastCommDate;
    public String type;
    public String state;
    public String activityState;
    public Long lastSyncDate;
    public String syncStatus;
    public Gateway gateway;
    public Data data;
    public Collection<Application> applications;
    public Map<String, MqttCommunication> communication;

    public static class Data {
        public Double rssi;
        public String rssiLevel;
        public String networkServiceType;
        public Double latitude;
        public Double longitude;
    }

    public static class Gateway {
        public String uid;
        public String imei;
        public String macAddress;
        public String serialNumber;
        public String type;
    }

}

File: mainActivity/src/main/java/net/airvantage/model/AvSystem.kt
package net.airvantage.model

import net.airvantage.utils.Predicate


class AvSystem {
    var uid: String? = null
    var name: String? = null
    var type: String? = null
    var state: String? = null
    var gateway: Gateway? = null
    var data: Data? = null
    var applications: MutableList<Application>? = null
    var communication: MutableMap<String, MqttCommunication>? = null

    class Data {
        var rssi: Double? = null
        var rssiLevel: String? = null
        var networkServiceType: String? = null
        var latitude: Double? = null
        var longitude: Double? = null
    }

    class Gateway {
        var uid: String? = null
        var imei: String? = null
        var serialNumber: String? = null
        var type: String? = null
    }

    companion object {


        fun hasSerialNumber(serialNumber: String?): Predicate<AvSystem> {
            return  { item -> serialNumber != null && serialNumber == item.gateway!!.serialNumber }
        }
    }


}
